src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -234,6 +235,26 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
                 // been applied to the result; make sure it's also successful.
                 ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
 
+                if (responseContent == null)
+                {
+                    Type createdODataResultType = actionExecutedContext.Result.GetType().GetGenericArguments().Count() > 0 ?
+                        typeof(CreatedODataResult<>).MakeGenericType(actionExecutedContext.Result.GetType().GetGenericArguments()[0]) : null;
```

Comment:
What happens if the result is neither an `Ok(result)` or `Created(result)`?

Reply:
Why do we create the CreatedODataResult<T> if we have generic arguments? Is it certain?

Reply:
@habbes Most of the results that we return  are a subclass of `ObjectResult` so they will be handled correctly